.github/workflows/build.yml

```diff
@@ -0,0 +1,61 @@
+# Initiate a github actions-initiated build
+name: Build and upload artifact on push to any branch
+on:
+  push: {}
+  workflow_dispatch: {}
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+      checks: write
+      issues: read
+      pull-requests: write
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Prepare Node
+      uses: actions/setup-node@v3
+      with:
+        node-version: 16.x
+    # Seed the build number with last number from TeamCity.
+    - name: Update GITHUB_RUN_NUMBER
+      run: |
+        LAST_TEAMCITY_BUILD=59
+        echo GITHUB_RUN_NUMBER=$(( $GITHUB_RUN_NUMBER + $LAST_TEAMCITY_BUILD )) >> $GITHUB_ENV
+    # now following example at https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs#example-using-yarn
+    - name: Install dependencies with yarn
+      run: yarn
+    - name: Clean and build target
+      run: |
+        yarn run clean
+        yarn run build
+    - name: Copy package.json to target
+      run: cp package.json target
+    - name: Go to target and install everything there in prod mode
+      run: |
+        cd target
+        yarn --production
+    - name: Head back up a dir
+      run: cd ..
+    - name: Make a dist dir and create a .zip file there
+      run: |
+        mkdir dist
+        zip -r dist/kindle-publication-check.zip target
```

Comment:
when I did this on https://github.com/guardian/formstack-submitter-ts/pull/31/files, I ended up with a zipfile that did not run in Lambda because the js files were under a `target` directory. Instead, I had to `cd` into the target directory and zip from there so the files are at the root

Reply:
Done. Will update main PR body with details